### ISSUE TITLE
Issues with OfflineAudioContext

### ISSUE THREAD

**Issue by MilesMorel** [magenta/magenta-js#436]
It looks like magenta is re-using a single OfflineAudioContext for processing in Onsets and Frames (and maybe elsewhere). This is fine if you only process a single blob and can do so synchronously (like in the demo), but this is problematic in the real-time streaming case. I think best practice is to create a new OfflineAudioContext every time you use it. This should prevent errors like "DOMException: cannot startRendering when an OfflineAudioContext is running" if you try to process from a stream.

Since the inference model is real-time, it would be a shame to have the model written in a way that can't be used with streams efficiently. Please let me know if I am mistaken here and thanks for sharing all this awesome tech!

**Comment by mdubey96** [comment]
Hey magenta team,

Thanks for developing such great technology!
Was wondering about this same issue and if there's a way to get this model working in real time?

**Comment by promagic007** [comment]
Hey magenta team,

Can you guide us how we can integrate real-time Onset and Frames?
Thanks.

**Comment by MilesMorel** [comment]
I decided to play with this again and got it working async and real-time without the OfflineAudioContext. I had to fork the project and change a few things that are perhaps specific to my use-case so I'll just describe what I did... The trick is to do the mel spec yourself with raw chunks (instead of an audiobuffer which is unnecessary) and then use `transcribeFromMelSpec` to avoid using the audio context in `preprocessAudio` altogether. I'd also recommend handling the model output yourself since the `pianorollToNoteSequence` implementation is too slow for real-time.
